images/ansible-operator/Dockerfile

```diff
@@ -11,20 +11,22 @@ RUN rustc --version
 
 # Copy python dependencies (including ansible) to be installed using Pipenv
 COPY images/ansible-operator/Pipfile* ./
+
 # Instruct pip(env) not to keep a cache of installed packages,
 # to install into the global site-packages and
 # to clear the pipenv cache as well
-ENV PIP_NO_CACHE_DIR=1 \
-    PIPENV_SYSTEM=1 \
-    PIPENV_CLEAR=1
+ENV PIP_NO_CACHE_DIR=1 
+
 # Ensure fresh metadata rather than cached metadata, install system and pip python deps,
 # and remove those not needed at runtime.
 RUN set -e && yum clean all && rm -rf /var/cache/yum/* \
   && yum update -y \
-  && yum install -y libffi-devel openssl-devel python39-devel gcc python39-pip python39-setuptools \
-  && pip3 install --upgrade pip~=23.3.2 \
-  && pip3 install pipenv==2023.11.15 \
-  && pipenv install --deploy \
```

Comment:
Pipenv usage seems to contradict by installing packages to global path and also, blocks usage of different version when used with `requires` version. As the docker provides isolation, using pip installer performs the same installation on the docker layers without conflicts.

Reply:
The following issue and PRs provide a context as to why Pipfile and Pipfile.lock were added:
- https://github.com/operator-framework/operator-sdk/issues/4237
- https://github.com/operator-framework/operator-sdk/pull/4494
- https://github.com/operator-framework/operator-sdk/pull/4543
The idea is to have fixed versions of the dependencies as well through the Pipfile and Pipfile.lock so that it can be locally reproduced.

Reply:
The point here is about the conflicts with respect to package installation path. The package management is still being managed by pipenv with usage of pipfile and pipfile.lock. Using  PIPENV_SYSTEM=1 enables installing the packages outside the virtual env set by pipenv. This resorts to using underlying system's python version than the desired version with which virtual env is instantiated.